src/main/java/me/zeroeightsix/fiber/annotation/Setting.java

```diff
@@ -1,33 +1,58 @@
 package me.zeroeightsix.fiber.annotation;
 
+import javax.annotation.RegEx;
+import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
 
 @Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.FIELD)
 public @interface Setting {
 
     boolean constant() default false;
     String name() default "";
     String comment() default "";
 
     boolean ignore() default false;
+
+    @Target(ElementType.FIELD)
     @Retention(RetentionPolicy.RUNTIME)
     @interface Node {
         String name() default "";
     }
 
-    @Retention(RetentionPolicy.RUNTIME)
+    @Target({})
     @interface Constrain {
 
+        @Target(ElementType.FIELD)
         @Retention(RetentionPolicy.RUNTIME)
         @interface Min {
             double value();
         }
 
+        @Target(ElementType.FIELD)
         @Retention(RetentionPolicy.RUNTIME)
         @interface Max {
             double value();
         }
 
+        @Target(ElementType.FIELD)
+        @Retention(RetentionPolicy.RUNTIME)
+        @interface MinStringLength {
+            int value();
+        }
+
+        @Target(ElementType.FIELD)
+        @Retention(RetentionPolicy.RUNTIME)
+        @interface MaxStringLength {
+            int value();
+        }
```

Comment:
While I'm not exactly _against_ the naming of these, they could be shorter. We're dealing with already very long, verbose names (`Setting.Constrain.MaxStringLength`).
Basically, anything dealing with constraining length is *probably* a string, so can I propose renaming this to `MinLength` and `MaxLength` for brevity?

Reply:
Wouldn't it imply that this annotation can work on collections too ?

Reply:
You're right. Good point.